envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoySnapshotFactory.kt

```diff
@@ -89,6 +89,7 @@ class EnvoySnapshotFactory(
         }
 
         return addRemovedClusters(previousClusters, currentClusters)
+            .toSortedMap(Comparator.naturalOrder())
```

Comment:
Why we need `Comparator.naturalOrder()`? I think it's a default one.

Reply:
Yes, it is. I wanted to make it clear that we need `naturalOrder` and if a default would change we won't break.

Reply:
I think we won't break if the default changes, because we are not interested in _how_ the entries are sorted, we only require them to be sorted _consistently_.
Side note: there is really no chance that the default will change. It would be a breaking change and there is really no other candidate to set as a default.

But I see no problem with stating it explicitly, so for me we can keep it.